### PR TITLE
refactor: cleanup bicep

### PR DESCRIPTION
**Pull request by xzf0587**

### CODE REVIEW

**Review by sffamily — DISMISSED** [review]

**Review by jayzhang — APPROVED** [review]

File: packages/fx-core/src/component/driver/arm/util/handleError.ts [hunk 1]
```diff
@@ -0,0 +1,226 @@
...
+        error = fetchInnerError(error);
+      }
+      if (error.code === "ResourceGroupNotFound") {
+        return err(
+          new ResourceGroupNotExistError(
+            deployCtx.resourceGroupName,
+            deployCtx.client.subscriptionId
+          )
+        );
+      } else if (error.code === "InvalidTemplate" || error.code === "InvalidTemplateDeployment") {
```
Comment (wenytang-ms) [inline packages/fx-core/src/component/driver/arm/util/handleError.ts:59]: this "invalidtemplate" "invalidatetemplatedeployment" or "resourcegroupNotfound" may abstract contant strings. you may submit another PR later.

File: packages/fx-core/src/component/driver/arm/util/handleError.ts [hunk 2]
```diff
@@ -0,0 +1,226 @@
...
+  export function formattedDeploymentError(deploymentError: any): any {
+    if (deploymentError.subErrors) {
+      const result: any = {};
+      for (const key in deploymentError.subErrors) {
+        const subError = deploymentError.subErrors[key];
+        if (subError.inner) {
+          result[key] = formattedDeploymentError(subError.inner);
+        } else {
+          const needFilter =
+            subError.error?.message?.includes("Template output evaluation skipped") &&
```
Comment (wenytang-ms) [inline packages/fx-core/src/component/driver/arm/util/handleError.ts:214]: same here

File: packages/fx-core/src/component/driver/arm/util/innerHandleError.ts [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+export async function innerGetDeploymentOperations(
+  client: ResourceManagementClient,
+  resourceGroupName: string,
+  deploymentName: string
+): Promise<DeploymentOperation[]> {
+  const res = [];
+  for await (const page of client.deploymentOperations
+    .list(resourceGroupName, deploymentName)
+    .byPage({ maxPageSize: 100 })) {
```
Comment (wenytang-ms) [inline packages/fx-core/src/component/driver/arm/util/innerHandleError.ts:26]: same here

File: packages/fx-core/src/component/utils/depsChecker/bicepChecker.ts [hunk 4]
```diff
@@ -96,61 +46,26 @@ export async function ensureBicepForDriver(
...
-  private readonly _version: string | undefined;
+  private readonly version: string;
   private readonly _axios: AxiosInstance;
 
-  constructor(logger?: LogProvider, telemetry?: TelemetryReporter, version?: string) {
+  constructor(version: string, logger?: LogProvider, telemetry?: TelemetryReporter) {
     this._logger = logger;
     this._telemetry = telemetry;
-    this._version = version;
+    this.version = version;
```
Comment (wenytang-ms) [inline packages/fx-core/src/component/utils/depsChecker/bicepChecker.ts:61]: it'e better to keep same format

**Review by wenytang-ms — APPROVED** [review]

**Review by sffamily — APPROVED** [review]

**Pull request merged**